### ISSUE TITLE
false-positive visibility change

### ISSUE THREAD

**Issue by rhaschke** [lvc/abi-compliance-checker#97]
I observe a [false-positive](https://travis-ci.org/ros-visualization/rviz/jobs/578880947#L1501-L1509) visibility change, although the [corresponding source file](https://github.com/ros-visualization/rviz/blob/melodic-devel/src/rviz/panel_dock_widget.h) didn't even changed between dumps. The [report](https://gist.githubusercontent.com/rhaschke/549295b8a7084237c1811b23d7654820/raw/ed87a815326a4a8a37760f53aea87af525294393/librviz.html) as well as [old](https://gist.githubusercontent.com/rhaschke/549295b8a7084237c1811b23d7654820/raw/ed87a815326a4a8a37760f53aea87af525294393/old_dump.txt) and [new](https://gist.githubusercontent.com/rhaschke/549295b8a7084237c1811b23d7654820/raw/ed87a815326a4a8a37760f53aea87af525294393/new_dump.txt) dumps can be found in [this gist](https://gist.github.com/rhaschke/549295b8a7084237c1811b23d7654820).
The source repository is: https://github.com/ros-visualization/rviz

Change | Effect
-- | --
This method became private. | Recompilation of a client program may be broken with the error message: '_ZN4rviz15PanelDockWidget7setIconE5QIcon' is private.
This method became private. | Recompilation of a client program may be broken with the error message: '_ZN4rviz15PanelDockWidget14setWindowTitleE7QString' is private.

The offending methods are still public in the source. Do you have any idea, what happens here?

**Comment by rhaschke** [comment]
I think, the culprit is the abi-dumper as a diff of both dumps reveals that these method are considered private in the newer version:
```sh
$ grep -B 4 -A 15 "'_ZN4rviz15PanelDockWidget7setIconE5QIcon'," *.dump
new.dump-                            '545060' => {
new.dump-                                          'Class' => '545034',
new.dump-                                          'Header' => 'panel_dock_widget.h',
new.dump-                                          'Line' => '57',
new.dump:                                          'MnglName' => '_ZN4rviz15PanelDockWidget7setIconE5QIcon',
new.dump-                                          'Param' => {
new.dump-                                                       '0' => {
new.dump-                                                                'name' => 'this',
new.dump-                                                                'type' => '582584'
new.dump-                                                              },
new.dump-                                                       '1' => {
new.dump-                                                                'name' => 'icon',
new.dump-                                                                'type' => '546535'
new.dump-                                                              }
new.dump-                                                     },
new.dump-                                          'Private' => 1,
new.dump-                                          'Return' => '1',
new.dump-                                          'ShortName' => 'setIcon',
new.dump-                                          'Source' => 'panel_dock_widget.cpp',
new.dump-                                          'SourceLine' => '83'
--
old.dump-                            '10985604' => {
old.dump-                                            'Class' => '10985311',
old.dump-                                            'Header' => 'panel_dock_widget.h',
old.dump-                                            'Line' => '57',
old.dump:                                            'MnglName' => '_ZN4rviz15PanelDockWidget7setIconE5QIcon',
old.dump-                                            'Param' => {
old.dump-                                                         '0' => {
old.dump-                                                                  'name' => 'this',
old.dump-                                                                  'type' => '10989931'
old.dump-                                                                },
old.dump-                                                         '1' => {
old.dump-                                                                  'name' => 'icon',
old.dump-                                                                  'type' => '535896'
old.dump-                                                                }
old.dump-                                                       },
old.dump-                                            'Return' => '1',
old.dump-                                            'ShortName' => 'setIcon',
old.dump-                                            'Source' => 'panel_dock_widget.cpp',
old.dump-                                            'SourceLine' => '83'
old.dump-                                          },
```
